### DISCUSSION TITLE
How to use Docker Desktop inside devcontainer

### DISCUSSION THREAD

**Discussion by RollsChris** [devcontainers/community#137] [Q&A]
Hi i'm looking to run the command `docker init ` inside my dev container, but that command can not be found.
I am using docker-outside-docker which gives me the `docker cli` tool but not the `init` command.

How can i get access to it? 

running Docker version i get this: 

Client:
 Version:           26.1.1-1
 API version:       1.45
 Go version:        go1.21.9
 Git commit:        4cf5afaefa0bec8ab94fcf631add24e284d0ecbf
 Built:             Tue Apr 30 11:40:03 UTC 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Desktop 4.29.0 (145265)
 Engine:
  Version:          26.0.0
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.8
  Git commit:       8b79278
  Built:            Wed Mar 20 15:18:01 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.28
  GitCommit:        ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

Thanks

**Comment by samruddhikhandale** [comment]
Hi 👋 

Docker Desktop itself cannot be accessed inside a dev container as it is a graphical user interface application. Hence, you can not access the `docker init` command which is specific to Docker Desktop. However, you can access the Docker daemon from inside a dev container, which allows you to run Docker commands.

This is typically done by mounting the Docker socket from the host into the dev container ; which you are currently achieving with the `docker-outside-docker` Feature.

**Reply by RollsChris** [reply]
i thought by using docker-outside-docker it had access to docker desktop on the host.

its weird that there are different Management commands between my dev container and wsl host.
